src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -207,10 +231,27 @@ export function createReactNodeViewConstructor(
       />
     );
 
+    if (contentDOM) {
+      // Store this node view's registration key on the contentDOM
+      // so it can be retrieved by its children.
+      contentDOM.dataset[REACT_PM_KEY] = key;
+    }
+
     const unregisterElement = registerElement(
+      // We register the element with the nearest ancestor's registration key
+      () => {
```

Comment:
If there's a reason this is a callback rather than just as tring argument, can you please remind me?

Reply:
I think we talked about something with the `editorView` needing to be done updating to be sure that `domAtPos` can find an element at the given position. I'm just surprised because I didn't understand function state setters in React to be anything but synchronously invoked.

Reply:
If that's it, I wonder if we should have our own little microtask in this file that we ensure is queued that can register all unregistered node views within a batch update.

Reply:
But that's maybe kinda what's happening here, but via React's state setting machinery, maybe?

Reply:
Like we're gonna call setters one or more times and React will take care of scheduling an update as a result.

Reply:
Except we're already inside a layout effect creation at this point, so I think each state set might be synchronous, which is kinda why I wonder if it'd be better to explicitly batch them.

Reply:
Yeah, it's because when we call `registerElement`, ProseMirror is still descending into the node view tree and hasn't assigned view.docView yet, so this call to domAtPos would fail. Not my favorite.

Reply:
Ah sorry GitHub hadn't shown me this whole thread when I responded. I think I'm into the idea of explicitly batching/queuing this work, but I'm not sure I have a clear picture of what that would actually look like as you're describing it!